internal/wazeroir/operations.go

```diff
@@ -1171,22 +1204,26 @@ func (OperationSet) Kind() OperationKind {
 	return OperationKindSet
 }
 
-// OperationGlobalGet implements Operation.
+//// OperationGlobalGet implements Operation.
+////
+//// The engines are expected to read the global value specified by OperationGlobalGet.Index,
+//// and push the copy of the value onto the stack.
+////
+//// See wasm.OpcodeGlobalGet.
+//type OperationGlobalGet struct{ Index uint32 }
 //
-// The engines are expected to read the global value specified by OperationGlobalGet.Index,
-// and push the copy of the value onto the stack.
+//// String implements fmt.Stringer.
+//func (o OperationGlobalGet) String() string {
+//	return fmt.Sprintf("%s %d", o.Kind(), o.Index)
+//}
 //
-// See wasm.OpcodeGlobalGet.
-type OperationGlobalGet struct{ Index uint32 }
-
-// String implements fmt.Stringer.
-func (o OperationGlobalGet) String() string {
-	return fmt.Sprintf("%s %d", o.Kind(), o.Index)
-}
+//// Kind implements Operation.Kind
+//func (OperationGlobalGet) Kind() OperationKind {
+//	return OperationKindGlobalGet
+//}
 
-// Kind implements Operation.Kind
-func (OperationGlobalGet) Kind() OperationKind {
-	return OperationKindGlobalGet
+func NewOperationGlobalGet(index uint32) OperationUnion {
+	return OperationUnion{OpKind: OperationKindGlobalGet, Us: []uint64{uint64(index)}}
```

Comment:
although we are obviously giving up type-safety, it may be still worth to introduce a named constructor for readability and to avoid mistakes. Consider commit https://github.com/tetratelabs/wazero/pull/1296/commits/34861a200eed3bf1f5b196093f3f21a7057f5680 -- at the beginning I forgot to explicitly set `OpKind` and everything defaulted to `Unknown` :D